server/src/main/java/com/defold/extender/services/DefoldSdkService.java

```diff
@@ -126,51 +117,112 @@ public CompletableFuture<DefoldSdk> getRemoteSdk(String hash) {
         return CompletableFuture.supplyAsync(() -> {
             long methodStart = System.currentTimeMillis();
             // Define SDK directory for this version
-            File sdkDirectory = new File(baseSdkDirectory.toFile(), hash);
+            File sdkDirectory = new File(this.configuration.getLocation().toFile(), hash);
             File sdkRootDirectory = new File(sdkDirectory, "defoldsdk");
             DefoldSdk sdk = new DefoldSdk(sdkRootDirectory, hash, this);
+            boolean isVerified = false;
             // If directory does not exist, create it and download SDK
-            if (!Files.exists(sdkDirectory.toPath())) {
+            if (Files.exists(sdkDirectory.toPath())) {
+                // if sdk already exists in cache - it means that sdk already verified
+                isVerified = true;
+            } else  {
                 boolean sdkFound = false;
-                for (String url_pattern : REMOTE_SDK_URL_PATTERNS) {
+                String url = null;
+                ClientHttpRequestFactory clientHttpRequestFactory = new SimpleClientHttpRequestFactory();
+                for (String urlPattern : configuration.getSdkUrls()) {
                     try {
-                        URL url = URI.create(String.format(url_pattern, hash)).toURL();
```

Comment:
Avoid conversion to `URL` because `ClientHttpRequestFactory::createRequest` takes `URI` object instead of `URL`. Helps remove `URISyntaxException`